### PR TITLE
Add inline docs for all public functions

### PR DESCRIPTION
**Pull request by Gudahtt**
All public functions now have inline documentation. Some of these descriptions are rather sparse, but it's a start. We can improve them later.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -396,6 +396,16 @@ export function normalize(input: number | string): string {
   return ethUtil.addHexPrefix(input.toLowerCase());
 }
 
+/**
+ * Create an Ethereum-specific signature for a message.
+ *
+ * This function is equivalent to the `eth_sign` Ethereum JSON-RPC method as specified in EIP-1417,
+ * as well as the MetaMask provider `personal_sign` method.
```
Comment (rekmarks) [inline src/index.ts:403]: I am on a crusade to make "provider" a proscribed term, at least until we can agree on what it means!

```suggestion
 * as well as the MetaMask's `personal_sign` method.
```

File: src/index.ts [hunk 2]
```diff
@@ -416,18 +435,44 @@ export function recoverPersonalSignature<T extends MessageTypes>(
   return senderHex;
 }
 
+/**
+ * Recover the public key of the account used to  create the given Ethereum signature. The message
```
Comment (rekmarks) [inline src/index.ts:439]:
```suggestion
 * Recover the public key of the account used to create the given Ethereum signature. The message
```

**Review by rekmarks — CHANGES_REQUESTED** [review]
Much-needed!

File: src/index.ts (hunk 1 above)
Comment (Gudahtt) [reply src/index.ts:403]: Sure! This still reads fine without that term.

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**